.github/workflows/Deploy.yaml

```diff
@@ -27,49 +31,101 @@ on:
+            $directCommit = '${{ github.event.inputs.directCommit }}'
+            $requireEndToEndTests = '${{ github.event.inputs.requireEndToEndTests }}'
+            $createRelease = '${{ github.event.inputs.createRelease }}'
+            $defaultBcContainerHelperVersion = '${{ github.event.inputs.defaultBcContainerHelperVersion }}'
+          }
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "branch=$branch"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "copyToMain=$copyToMain"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "directCommit=$directCommit"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "requireEndToEndTests=$requireEndToEndTests"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "createRelease=$createRelease"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "defaultBcContainerHelperVersion=$defaultBcContainerHelperVersion"
+
   CheckEndToEnd:
     runs-on: [ ubuntu-latest ]
+    needs: [ Inputs ]
     steps:
       - name: Check successful end 2 end tests have run
-        if: github.repository_owner == 'microsoft' && github.event.inputs.requireEndToEndTests == 'true'
+        if: github.repository_owner == 'microsoft' && needs.Inputs.outputs.requireEndToEndTests == 'true'
         env:
           GH_TOKEN: ${{ github.token }}
         run: |
           $errorActionPreference = "Stop"
-          $end2endRuns = (gh api /repos/$($env:GITHUB_REPOSITORY)/actions/runs?per_page=100 | ConvertFrom-Json).workflow_runs | Where-Object { $_.name -eq 'End 2 End Tests' }
+          $end2endRuns = (gh api /repos/$($env:GITHUB_REPOSITORY)/actions/workflows/E2E.yaml/runs?per_page=100 | ConvertFrom-Json).workflow_runs
           $latestSha = (gh api /repos/$($env:GITHUB_REPOSITORY)/commits/main | ConvertFrom-Json).sha
-          $latestRun = $end2endruns | Where-Object { $_.head_sha -eq $latestSha }
+          $latestRun = $end2endruns | Where-Object { $_.Name -eq "End to end tests - $latestSha" } | select-object -first 1
           if (!$latestRun) {
-            throw "No End 2 End Tests run found for the latest commit on main"
+            throw "No End to end tests run found for the latest commit on main"
           }
           if ($latestRun.status -ne 'completed') {
-            throw "End 2 End Tests run for the latest commit on main is not completed"
+            throw "End to end tests run for the latest commit on main is not completed - see $($latestRun.html_url)"
           }
           if ($latestRun.conclusion -ne 'success') {
-            throw "End 2 End Tests run for the latest commit on main did not succeed"
+            throw "End to end tests run for the latest commit on main did not succeed - see $($latestRun.html_url)"
+          }
+          $allJobs = (gh api /repos/$($env:GITHUB_REPOSITORY)/actions/runs/$($latestRun.id)/jobs | ConvertFrom-Json).jobs
```

Comment:
This would return only 30 jobs by default.

Alexander used `--paginate` in his PR: https://github.com/microsoft/AL-Go/blob/f82966816d304985bde80f66e82e5b36b0673141/Actions/WorkflowPostProcess/WorkflowPostProcess.ps1#L18

Reply:
good catch

Reply:
How was this addressed?

Reply:
by adding --paginate - but that was undone when trying to solve the other comment, thanks

Reply:
redone